src/Commands/MakeModelsCommand.php

```diff
@@ -118,9 +118,24 @@ protected function getSchemaTables()
                 $filterTablesWhere = ' AND table_name IN (\'' . implode('\', \'', $tableNamesToFilter) . '\')';
             }
         }
-
-        $tables = \DB::select("SELECT table_name AS `name` FROM information_schema.tables WHERE table_schema = DATABASE()" . $filterTablesWhere);
-
+        
+        switch (env('DB_CONNECTION')) {
+           case 'mysql':
+               $tables = \DB::select("SELECT table_name AS name FROM information_schema.tables WHERE table_type='BASE TABLE' AND table_schema = '" . env('DB_DATABASE') . "'" . $filterTablesWhere);
+               break;
+               
+           case 'sqlsrv' || 'dblib':
+               $tables = \DB::select("SELECT table_name AS name FROM information_schema.tables WHERE table_type='BASE TABLE' AND table_catalog = '" . env('DB_DATABASE') . "'" . $filterTablesWhere);   
+               break;
+           
+           case 'pgsql':            
+               $tables = \DB::select("SELECT table_name AS name FROM information_schema.tables WHERE table_schema = 'public' AND table_type='BASE TABLE' AND table_catalog = '" . env('DB_DATABASE') . "'" . $filterTablesWhere);   
+               
+           default:
+               $tables = [];
+               echo env("DB_CONNECTION") . ': Unsupported database';
```

Comment:
That's right. 
I have removed the default condition. Excepcion error thrown when invalid database is configured on .env file.